src/main/java/com/conveyal/datatools/manager/gtfsplus/GtfsPlusValidation.java

```diff
@@ -299,7 +318,26 @@ private static void validateTableValue(
                 }
                 break;
         }
+    }
+
+    /** Validate a single route_id value for the directions.txt GTFS+ table. */
+    private static void validateTableValue(
+            Collection<ValidationIssue> issues,
+            String tableId,
+            int rowIndex,
+            String[] allValues,
+            String value,
+            JsonNode[] specFieldsFound,
+            JsonNode specField,
+            GTFSFeed gtfsFeed,
+            Map<String, Route> gtfsRoutes
+    ) {
+        if (specField == null) return;
+        validateTableValue(issues, tableId, rowIndex, allValues, value, specFieldsFound, specField, gtfsFeed);
 
+        if (!gtfsRoutes.containsKey(value)) return;
+        // "Check off" the route_id from the list to verify every route id has a direction
+        gtfsRoutes.remove(value);
```

Comment:
This overloading method probably isn't needed. You can just pass the gtfsRoutes into the original method and add this line at the bottom of it.